### DISCUSSION TITLE
Open `.csv` file in text mode (Related to #2187)

### DISCUSSION THREAD

**Discussion by kylebutts** [posit-dev/positron#5198] [Feedback] [answered]
The most recent versions of positron now automatically open csv files in headless mode (via the new positron/duckdb extension) which is definitely an exciting feature. But I can not figure out how to open up a csv as a plain text file which also has advantages (e.g. editing them). Is this possible currently? If not, it would be nice to have

**Comment by juliasilge** [comment, marked as answer]
You can right click and open it with the text editor:

![Screenshot 2024-10-29 at 10 07 48 AM](https://github.com/user-attachments/assets/e3865b07-1d22-4344-ad7e-cb7df59e1859)

Side note: you may enjoy the [Rainbow CSV](https://open-vsx.org/extension/mechatroner/rainbow-csv) extension. 🌈

**Reply by kylebutts** [reply]
Ahh, `open with...`!

**Comment by jmcphers** [comment]
I think we should also add a button to the data explorer to take you to the CSV. Filed that here: https://github.com/posit-dev/positron/issues/5206

**Comment by mattkerlogue** [comment]
Oh, I've only just noticed this here, sorry forgot to check the discussions before filing #11057 which is related to this request.

The update in #6132 is a useful addition to get from the data viewer to the plain text, but I still think it would be useful to have a clearer option in the context menu itself. It's really not intuitive that 'open with…' leads you to an option within Positron to select between the text editor and data explorer, instead I routinely think it's going to give me the option to open in an external application (like the 'open with…' option in Mac OS Finder's context menu).
